Source/DafnyCore/Auditor/Assumption.cs

```diff
@@ -4,56 +4,90 @@
 
 namespace Microsoft.Dafny.Auditor;
 
-public record AssumptionDescription(string issue, string mitigation, bool isExplicit) {
+public record AssumptionDescription(
+  string issue,
+  string mitigation,
+  string mitigationIETF,
+  bool isExplicit
+) {
   public static AssumptionDescription HasAxiomAttribute = new(
     issue: "Declaration has explicit [{:axiom}] attribute.",
     mitigation: "Provide a proof or test.",
+    mitigationIETF: "SHOULD provide a proof or test.",
     isExplicit: true);
   public static AssumptionDescription HasVerifyFalseAttribute = new(
     issue: "Declaration has [{:verify false}] attribute.",
     mitigation: "Remove and prove if possible.",
+    mitigationIETF: "MUST remove and prove.",
+    isExplicit: false);
+  public static AssumptionDescription ExternFunction = new(
+    issue: "Function with [{:extern}] attribute.",
+    mitigation: "Turn into a [method] or test thoroughly for lack of side effects.",
```

Comment:
a method with `modifies *` ?
Personally, using a "function" instead of a "method" for externs is just making two more assumptions: determinism w.r.t. inputs and no side-effects.
Would that be better if we observed one more attribute, e.g. {:axiom} ?

Reply:
Yeah, I think an `{:axiom}` annotation would make it clear that it's an intentional assumption.

So I'll update that message to suggest adding `modifies *` and testing for determinism, as well as potentially adding `{:axiom}`.